motoko/hello-angular-motoko/README.md

```diff
@@ -0,0 +1,114 @@
+# hello-angular-motoko
+This is an example of a Motoko project with an Angular 14 frontend.
+
+<a name="prerequesites"></a> 
+## Prerequesites
+- [Install DFX](https://internetcomputer.org/docs/current/developer-docs/quickstart/local-quickstart).
+- [Install Angular Globally](https://angular.io/guide/setup-local)
+- Install npm packages from the project root:
+    ```
+    npm install
+    ```
+- A https://nns.ic0.app wallet, logged in.
+- ~5 US dollars worth of ICP transfered to your wallet address.
+- As taken from [this guide](https://kyle-peacock.com/blog/dfinity/your-first-canister): 
+ 
+    1. Login to your https://nns.ic0.app, select your account and verify your balance.
+    2. Select the [CANISTERS](https://nns.ic0.app/#/canisters) view on the left side navigation menu. 
+    3. Click the blue "Create or Link Canister" button at the bottom of your page to open up the dialog.
+    4. Click the "Create New Canister" button. Then, select the ICP account that you want to use to fund your new canister.
+    5. You'll then be prompted to enter an amount. For now, put at least 2T Cycles.
+    6. Click "Review Cycles Purchase" and Confirm.
+        * You will need to create 2 unique Canisters for this project. Repeat steps 3-6 to create a second canister. 
+    7. You'll need to get the principal from your computer by running from the project root
+        ```
+        dfx identity get-principal
+        ```
+    8. Copy that principal id and head back to your CANISTERS view with your newly created Canisters. Click the blue "Change Controllers" button to open up the controller management UI.
+        * Repeat for the second Canister.
+    9. Enter your principal in the empty second input, and then click "Perform Controller Change". This step authorizes your dfx on your computer to deploy to your newly created Canister. 
+        * Repeat for the second Canister as well.
+    10. Modify the `canister_id.json` file in the project root. Update both "motoko" and "www" "ic" properties.
+	 
+        ```bash
+        # canister_id.json
+        {
+        "motoko": {
+            # replace this with your unique cansiter from the above guide
+            "ic":"your-unique-canister-goes-here"
+        },
+        "www": {
+            # replace this with a second unique cansiter from the above guide
+            "ic": "another-unique-canister-goes-here"
+        }
+    
+        ```
```

Comment:
```suggestion
- For live IC deployment: A wallet set up as described in the [quickstart guide](https://internetcomputer.org/docs/current/developer-docs/quickstart/hello10mins/).
```
With a configured wallet (which I think is a reasonable demand), this could all be replaced by `dfx canister --network ic create --all --with-cycles 1000000000000`, which `dfx deploy --network ic` automatically does when you also add `--with-cycles 1000000000000`.

Reply:
@sesi200 
- How does the project know what canisters to deploy to? This is the recommended method as given by @krpeacock. I imagine it will do well for now.
- As you stated before, simply providing a link without explicitly describing the instructions is bad practice.

Reply:
I didn't explain myself enough the first time around, sorry. The reasons I didn't like the link to Kyle's article are:
- It's not the standard way to do it (anymore?). One dfx-command does the trick just as well.
- It links to Kyle's _personal_ website. While he's on the SDK team, the article is not an official (and therefore maintained) resource. By now it contains some outdated info (e.g. the URL to the SDK download), and the NNS interface shown in the screenshots is not accurate anymore.
- The link was combined with a bunch of deviations from the instructions. This makes it really hard to follow because people will have to switch between the two pages all the time.

Why I think the new link is fine:
- It's an official (and therefore maintained) resource.
- If we didn't link there, we'd be duplicating information, which makes maintenance much harder.
- It's literally where we send _everyone_ to start. We'd like to use at least that as common base knowledge and assume this in (almost?) all other examples.
- The requirements section is a _lot_ smaller this way.
- It's more clear that this is not a hard requirement, but only for live deployment. (Yes, this is only tangentially related to the link and more about how it's presented, but wanted to mention it anyways)

I hope that you can accept the new link with this reasoning.

(the question about which canister to deploy to is answered in a different comment)